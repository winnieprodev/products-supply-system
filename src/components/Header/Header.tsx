import { useScrollPosition } from "hooks";
import React from "react";
import { Empty } from "utils/types";

import * as Styled from "./header.styled";

const Header = ({ children }: React.PropsWithChildren<Empty>) => {
  const scrollPosition = useScrollPosition();
  return (
    <Styled.Header $scrolled={scrollPosition > 30}>{children}</Styled.Header>
  );
};

const AlignLeft = ({ children }: React.PropsWithChildren<Empty>) => (
  <Styled.AlignLeft>{children}</Styled.AlignLeft>
);

const AlignRight = ({ children }: React.PropsWithChildren<Empty>) => (
  <Styled.AlignRight>{children}</Styled.AlignRight>
);

Header.AlignLeft = AlignLeft;
Header.AlignRight = AlignRight;

export default Header;
