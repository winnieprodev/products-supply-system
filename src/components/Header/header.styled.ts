import styled, { css } from "styled-components";

interface HeaderProps {
  $scrolled: boolean;
}
export const Header = styled.div<HeaderProps>`
  position: sticky;
  top: 0;
  padding: 16px;
  background: white;
  display: grid;
  grid-template-columns: [left] 1fr [right];
  z-index: +1;

  ${({ $scrolled }) =>
    $scrolled &&
    css`
      box-shadow: 0 4px 10px rgba(0, 0, 0, 0.15);
    `}
`;

export const AlignLeft = styled.div`
  grid-area: left;
  align-self: center;
`;

export const AlignRight = styled.div`
  grid-area: right;
  align-self: center;
  display: flex;
  align-items: center;
`;
