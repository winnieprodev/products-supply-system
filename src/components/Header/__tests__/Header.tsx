import { render } from "@testing-library/react";

import Header from "../Header";

describe("Header", () => {
  describe("Snapshot tests", () => {
    test("it matches the snapshot for basic use", () => {
      const { container } = render(
        <Header>
          <Header.AlignLeft>Aligned to the left</Header.AlignLeft>
          <Header.AlignRight>Aligned to the right</Header.AlignRight>
        </Header>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
