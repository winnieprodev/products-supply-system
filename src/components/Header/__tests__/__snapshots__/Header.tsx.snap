// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Snapshot tests it matches the snapshot for basic use 1`] = `
.c0 {
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  padding: 16px;
  background: white;
  display: grid;
  grid-template-columns: [left] 1fr [right];
  z-index: +1;
}

.c1 {
  grid-area: left;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c2 {
  grid-area: right;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    Aligned to the left
  </div>
  <div
    class="c2"
  >
    Aligned to the right
  </div>
</div>
`;
